pkg/controllers/management/etcdbackupconfig/etcdbackupconfig.go

```diff
@@ -0,0 +1,179 @@
+}
+
+func (c *Controller) syncBackups(ctx context.Context, i time.Duration) {
+	// main backup config sync loop
+	for range ticker.Context(ctx, i) {
+		// get available backup configs
+		backupConfigs, err := c.backupConfigClient.List(metav1.ListOptions{})
+		if err != nil {
+			logrus.Infof("melsayed failed to list configs: %v", err)
+			continue
+		}
+
+		//Work on found configs
+		for _, bc := range backupConfigs.Items {
+			logrus.Infof("melsayed found backupConfig %s, lastSeen: %s", bc.Name, getCondition(&bc, "LastSeen").LastUpdateTime)
+			// set seen condition, probably don't need this.
+			if err := c.setSeenCondition(bc.Name); err != nil {
+				logrus.Errorf("melsayed failed to setSeen: %v", err)
+			}
+
+			// kick the backup "thing"
+
+			configDuration, err := time.ParseDuration(bc.Creation)
+			if err != nil {
+				logrus.Infof("melsayed can't parse duration %s : %v ", bc.Creation, err)
+				continue
+			}
+
+			if err := c.setLastStartedCondition(bc.Name); err != nil {
+				logrus.Errorf("melsayed failed to setLastStarted: %v", err)
+				continue
+			}
+			lastCompleted := getCondition(&bc, "LastCompleted")
+			if lastCompleted == nil {
+				logrus.Infof("melsayed first time backup")
+				c.kickBackup(bc.Name)
+				continue
+			}
+			lastCompletedTime, _ := time.Parse(time.RFC3339, lastCompleted.LastUpdateTime)
+			// It's been more than bc.Creation since we ran last successfull backup
+			if time.Since(lastCompletedTime) > configDuration {
+				c.kickBackup(bc.Name)
+			}
+		}
+	}
+
+}
+
+func (c *Controller) setSeenCondition(backupConfigName string) error {
+	bc, _ := c.getBackupConfig(backupConfigName)
+	setCondition(bc, v3.EtcdBackupConfigCondition{
+		Type:           "LastSeen",
+		Status:         "True",
+		LastUpdateTime: time.Now().Format(time.RFC3339),
+	})
+
+	return c.updateBackupConfig(bc)
+}
+
+func (c *Controller) setLastStartedCondition(backupConfigName string) error {
+	bc, err := c.getBackupConfig(backupConfigName)
+	if err != nil {
+		return err
+	}
+	setCondition(bc, v3.EtcdBackupConfigCondition{
+		Type:           "LastStarted",
+		Status:         "True",
+		LastUpdateTime: time.Now().Format(time.RFC3339),
+	})
+
+	return c.updateBackupConfig(bc)
+}
+
+func (c *Controller) setLastCompletedConddition(backupConfigName string) error {
+	bc, err := c.getBackupConfig(backupConfigName)
+	if err != nil {
+		return err
+	}
+	setCondition(bc, v3.EtcdBackupConfigCondition{
+		Type:           "LastCompleted",
+		Status:         "True",
+		LastUpdateTime: time.Now().Format(time.RFC3339),
+	})
+
+	return c.updateBackupConfig(bc)
+}
+
+func (c *Controller) getBackupConfig(name string) (*v3.EtcdBackupConfig, error) {
+	return c.backupConfigClient.Get(name, metav1.GetOptions{})
+}
+
+func (c *Controller) updateBackupConfig(bc *v3.EtcdBackupConfig) error {
+	_, err := c.backupConfigClient.Update(bc)
+	return err
+}
+
+func setCondition(backupConfig *v3.EtcdBackupConfig, newCondition v3.EtcdBackupConfigCondition) {
```

Comment:
we have a framework for setting the condition; you can check provisioner.go code to get more details on that.